### PR TITLE
:bug: Low-Code CDK: convert expires_in to int when refreshing oauth

### PR DESCRIPTION
**Pull request by joelluijmes**
## What
Converts expires_in to int when refreshing during oauth refresh. Some oauth providers return the expires_in as a string in the returned json. 

```
{
    "access_token": "....",
    "refresh_token": "....",
    "expires_in": "600",
}
```

Currently this is silently ignored, but when the expire date is calculated a type error is raised.

```
File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py", line 27, in get_auth_header
    return {"Authorization": f"Bearer {self.get_access_token()}"}
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py", line 195, in get_access_token
    self.set_token_expiry_date(t0.add(seconds=access_token_expires_in))
  File "/usr/local/lib/python3.9/site-packages/pendulum/datetime.py", line 632, in add
    dt = add_duration(
  File "/usr/local/lib/python3.9/site-packages/pendulum/helpers.py", line 113, in add_duration
    if abs(seconds) > 59:
TypeError: bad operand type for abs(): 'str'
```

## How
Always parse the value as int

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No, it un-breaks 🤓 

## Pre-merge Checklist
Checklists seems redundant for this change

- [x] updated unit tests

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/sources/streams/http/requests_native_auth/test_requests_native_auth.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def test_refresh_access_token(self, mocker):
 
         assert ("access_token", 1000) == token
 
+        # Test with expires_in as str
+        mocker.patch.object(resp, "json", return_value={"access_token": "access_token", "expires_in": "2000"})
+        token = oauth.refresh_access_token()
+
+        assert ("access_token", 2000) == token
+
```
Comment (joelluijmes) [inline airbyte-cdk/python/unit_tests/sources/streams/http/requests_native_auth/test_requests_native_auth.py:171]: Expanded the unit tests with another assertion. Not sure if this is conventional in Airbyte repo, or new test should be added. LMK.

**Comment by joelluijmes** [comment]
Also, I only updated the `requests_native_auth` classes. As the [other version](airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/oauth.py) is already marked deprecated.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def refresh_access_token(self) -> Tuple[str, int]:
         """
         try:
             response_json = self._get_refresh_access_token_response()
-            return response_json[self.get_access_token_name()], response_json[self.get_expires_in_name()]
+            return response_json[self.get_access_token_name()], int(response_json[self.get_expires_in_name()])
```
[the remark this answers is not in the document]
Comment (joelluijmes) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py:80]: `get_expires_in_name` only returns the name of the field in the response, which defaults to `expires_in`. It's not actually a function which parses the value.

Or are you suggesting to refactor the oauth classes to work in such way?

[the remark this answers is not in the document]
Comment (marcosmarxm) [reply airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py:80]: Oh I missed that :ops:

**Review by girarda — APPROVED** [review]
lgtm! :shipit:

**Review by brianjlai — APPROVED** [review]

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
Hello :wave::skin-tone-2: and thank you for your contribution!

Airbyte has instituted a code freeze between 19 and 30 December, to make sure there are no disruptions during the holidays. 
Because of this, reviewing and merging your contribution may take longer than usual. 
We apologize for the delay, but we want everyone to have a quiet and happy holiday.

If you have any questions or need further clarification, please don't hesitate to ping via Slack.

**Comment by joelluijmes** [comment]
This PR has been approved, is there something else I need to do before we can get this merged? 😇

**Comment by joelluijmes** [comment]
Updated with master again. Would be great if this can be merged 😇

**Comment by sherifnada** [comment]
@joelluijmes i had to revert this PR due to unit test failures. It will be re-merged as part of https://github.com/airbytehq/airbyte/pull/20301

**Pull request merged**